src/main/java/io/lettuce/core/cluster/ClusterPubSubConnectionProvider.java

```diff
@@ -28,7 +28,7 @@ class ClusterPubSubConnectionProvider<K, V> extends PooledClusterConnectionProvi
 
     private final RedisCodec<K, V> redisCodec;
 
-    private final RedisClusterPubSubListener<K, V> notifications;
+    private final RedisClusterShardedPubSubListener<K, V> notifications;
```

Comment:
What is the purpose of having another class for sharded pub/sub? Why can't we just extend the existing interface?